src/lib/containers/drug_upload_tool/UserFileGrid.tsx

```diff
@@ -0,0 +1,269 @@
+    })
+
+    try {
+      await SynapseClient.deleteEntity(token, entityId);
+      const fileList = await this.getFileListingInFolder(token, this.state.dataFolderId!)
+      this.setState({
+        fileList
+      })
+    }
+    catch (error) {
+      this.onError(error)
+      return [];
+    }
+    finally {
+      this.setState({
+        isLoading: false
+      })
+    }
+  }
+
+  /* ------------------------------------------   rendering fns  ------------------------------------------------*/
+ 
+  renderLoading = (token: string | undefined, isLoading: boolean): JSX.Element => {
+    if (
+      token &&
+      isLoading) {
+      return (<div className='text-center'>
+        <span>Loading&hellip;</span>
+        <span style={{ marginLeft: '2px' }} className={'spinner'} />
+      </div>)
+    } else {
+      return <></>
+    }
+  }
+
+  renderFileTable = (fileList: Entity[], pathpart: string, dataFolderId?: string): JSX.Element => {
+    if (!dataFolderId) {
+      this.onError('Data Folder ID is undefined')
+      return <></>
+    }
+    if (_.isEmpty(fileList)) {
+      return <h5 className='text-center no-submissions padding-full'>You don't have submissions in progress [COPY!]</h5>
+    }
+    return (<table className='table file-table' >
+      <thead>
+        <tr>
+          <th>Submission Name</th>
+          <th>Edited On</th>
+          <th>Submitted On</th>
+          <th>Action</th>
+
+        </tr>
+      </thead>
+      <tbody>
+        {fileList.map((entity, i: number) => {
+          return (<tr className='test' key={i}>
+            <td><Link className='nav-green' to={`/${pathpart}/${dataFolderId}/${entity.id}/`}>{entity.name}</Link></td>
+            <td>{moment(entity.modifiedOn).calendar()}</td>
+            <td>?</td>
+            <td><button className='btn' aria-label="delete" onClick={() => this.setState({ modalContext: { action: this.deleteFile, arguments: [this.state.token!, entity.id!] } })}><FontAwesomeIcon icon={faTrash} aria-hidden="true" ></FontAwesomeIcon></button></td>
+          </tr>)
+        })}
+      </tbody>
+    </table>
+    )
+  }
+
+  render() {
+
+    return (<div className='container'>
+      {this.renderLoading(this.state.token, this.state.isLoading)}
+
+      {!(this.state.isLoading) &&
+        <div>
+          <h3>Your Submissions</h3>
+          <div className='panel panel-default padding-full'>
+            {this.renderFileTable(this.state.fileList, this.props.pathpart, this.state.dataFolderId)}
+            <div className='text-center'>
+              <a className='btn btn-success' href={`/${this.props.pathpart}/${this.state.dataFolderId}`}>Add new Compound</a></div>
+          </div></div>}
+      {this.state.modalContext &&
+        <WarningModal show={true} title={this.modalTitle} copy={this.modalCopy} callbackArgs={this.state.modalContext.arguments} onCancel={() => this.setState({ modalContext: undefined })} onOK={(x: string, y: string) => this.deleteFile(x, y)}></WarningModal>}
```

Comment:
When we were using tslint there was a rule that when a component had a lot of props (~3+), to place them on their own line so its a bit more readable, I will look into eslint and see about enabling that back again.
```JSX
<WarningModal 
      show={true}
      title={this.modalTitle}
      copy={this.modalCopy}
      callbackArgs={this.state.modalContext.arguments} 
      onCancel={() => this.setState({ modalContext: undefined })} 
      onOK={(x: string, y: string) => this.deleteFile(x, y)}>
</WarningModal>
```